cmd/secret-wrapper/main.go

```diff
@@ -202,3 +207,23 @@ func createSecret(client coreclientset.SecretInterface, name, dir string, dry bo
 	}
 	return nil
 }
+
+// uploadKubeconfig will do a best-effort attempt at uploading a kubeconfig
+// file if one does not exist at the time we start running but one does get
+// created while executing the command
+func uploadKubeconfig(client coreclientset.SecretInterface, name, dir string, dry bool) {
+	if _, err := os.Stat(path.Join(dir, "kubeconfig")); err == nil {
+		// kubeconfig already exists, no need to do anything
+		return
+	}
+	if err := wait.PollInfinite(time.Second, func() (done bool, err error) {
+		if _, err := os.Stat(path.Join(dir, "kubeconfig")); err != nil {
+			return false, nil
+		}
+		// kubeconfig exists, we can upload it
+		uploadErr := createSecret(client, name, dir, dry)
```

Comment:
Can this race with the call in `run`?

Reply:
Also: is `kubeconfig` written atomically or can it happen that an intermediary state is used?

Reply:
The race would be:

 - process A reads secret from APIServer
 - process B reads data from APIServer
 - process B updates data on APIServer
 - process A fails to update data on APIServer (resource version too old)

The only case in which this could potentially be bad is if process B read the data from disk before process A did, and had a stale view of that data but managed to update the APIServer. I can add a mutex to serialize updates but I am struggling to think of a way the race would end up with a bad state in the secret.

re: atomic ... I have no idea, but I assume yes.

Reply:
Checked, it is atomic

Reply:
fixed the race